chopper/test/utils_test.dart

```diff
@@ -554,17 +933,702 @@ void main() {
+        () => expect(
+          mapToQuery(map, listFormat: ListFormat.comma),
+          query,
+        ),
+      ),
+    );
+  });
+
+  group('mapToQuery maps with comma with includeNullQueryVars', () {
+    <Map<String, dynamic>, String>{
+      {
+        'foo': {'bar': 'baz'},
+      }: 'foo%5Bbar%5D=baz',
+      {
+        'foo': {'bar': ''},
+      }: 'foo%5Bbar%5D=',
+      {
+        'foo': {'bar': null},
+      }: 'foo%5Bbar%5D=',
+      {
+        'foo': {'bar': ' '},
+      }: 'foo%5Bbar%5D=%20',
+      {
+        'foo': {'bar': '\t'},
+      }: 'foo%5Bbar%5D=%09',
+      {
+        'foo': {'bar': 'baz', 'etc': 'xyz', 'space': ' ', 'tab': '\t'},
+      }: 'foo%5Bbar%5D=baz&foo%5Betc%5D=xyz&foo%5Bspace%5D=%20&foo%5Btab%5D=%09',
+      {
+        'foo': {
+          'bar': 'baz',
+          'int': 123,
+          'double': 456.789,
+          'zero': 0,
+          'negInt': -123,
+          'negDouble': -456.789,
+          'emptyString': '',
+          'nullValue': null,
+          'space': ' ',
+          'tab': '\t',
+          'list': ['a', 123, false],
+        },
+      }: 'foo%5Bbar%5D=baz&foo%5Bint%5D=123&foo%5Bdouble%5D=456.789&foo%5Bzero%5D=0&foo%5BnegInt%5D=-123&foo%5BnegDouble%5D=-456.789&foo%5BemptyString%5D=&foo%5BnullValue%5D=&foo%5Bspace%5D=%20&foo%5Btab%5D=%09&foo%5Blist%5D=a%2C123%2Cfalse',
+      {
+        'foo': {'bar': 'baz'},
+        'etc': 'xyz',
+      }: 'foo%5Bbar%5D=baz&etc=xyz',
+      {
+        'foo': {
+          'bar': 'baz',
+          'zap': 'abc',
+          'etc': {
+            'abc': 'def',
+            'ghi': 'jkl',
+            'mno': {
+              'opq': 'rst',
+              'uvw': 'xyz',
+              'aab': [
+                'bbc',
+                'ccd',
+                'eef',
+              ],
+            },
+          },
+        },
+      }: 'foo%5Bbar%5D=baz&foo%5Bzap%5D=abc&foo%5Betc%5D%5Babc%5D=def&foo%5Betc%5D%5Bghi%5D=jkl&foo%5Betc%5D%5Bmno%5D%5Bopq%5D=rst&foo%5Betc%5D%5Bmno%5D%5Buvw%5D=xyz&foo%5Betc%5D%5Bmno%5D%5Baab%5D=bbc%2Cccd%2Ceef',
+      {
+        'filters': {
+          r'$or': [
+            {
+              'date': {
+                r'$eq': '2020-01-01',
+              }
+            },
+            {
+              'date': {
+                r'$eq': '2020-01-02',
+              }
+            }
+          ],
+          'author': {
+            'name': {
+              r'$eq': 'John doe',
+            },
+          }
+        }
+      }: 'filters%5B%24or%5D=%7Bdate%3A%20%7B%24eq%3A%202020-01-01%7D%7D%2C%7Bdate%3A%20%7B%24eq%3A%202020-01-02%7D%7D&filters%5Bauthor%5D%5Bname%5D%5B%24eq%5D=John%20doe',
+    }.forEach(
+      (map, query) => test(
+        '$map -> $query',
+        () => expect(
+          mapToQuery(
+            map,
+            listFormat: ListFormat.comma,
+            includeNullQueryVars: true,
+          ),
+          query,
+        ),
+      ),
+    );
+  });
+
+  group('mapToQuery maps with indices and nested lists', () {
```

Comment:
Add the following tests in order to cover the new `ListFormat` based formatting when encoding nested lists:

```dart

  group(
    'mapToQuery maps with repeat (default) and nested lists',
    () {
      <Map<String, dynamic>, String>{
        {
          'filters': {
            r'$or': [
              {
                'date': {
                  r'$eq': '2020-01-01',
                }
              },
              null,
              {
                'date': {
                  r'$eq': '2020-01-02',
                }
              }
            ],
            'author': {
              'name': {
                r'$eq': 'Kai doe',
              },
            }
          }
        }: r'filters%2E$or%2Edate.$eq=2020-01-01&filters%2E$or%2Edate.$eq=2020-01-02&filters%2Eauthor%2Ename.$eq=Kai%20doe',
        {
          'filters': {
            'id': {
              r'$in': [3, 6, 8],
            },
          }
        }: r'filters%2Eid%2E$in=3&filters%2Eid%2E$in=6&filters%2Eid%2E$in=8'
      }.forEach(
        (map, query) {
          test(
            '$map -> $query',
            () => expect(
              mapToQuery(
                map,
              ),
              query,
              reason: 'legacy default',
            ),
          );

          test(
            '$map -> $query',
            () => expect(
              mapToQuery(
                map,
                listFormat: ListFormat.repeat,
              ),
              query,
            ),
          );
        },
      );
    },
  );

  group(
    'mapToQuery maps with brackets and nested lists',
    () {
      <Map<String, dynamic>, String>{
        {
          'filters': {
            r'$or': [
              {
                'date': {
                  r'$eq': '2020-01-01',
                }
              },
              {
                'date': {
                  r'$eq': '2020-01-02',
                }
              }
            ],
            'author': {
              'name': {
                r'$eq': 'Kai doe',
              },
            }
          }
        }: 'filters%5B%24or%5D%5B%5D%5Bdate%5D%5B%24eq%5D=2020-01-01&filters%5B%24or%5D%5B%5D%5Bdate%5D%5B%24eq%5D=2020-01-02&filters%5Bauthor%5D%5Bname%5D%5B%24eq%5D=Kai%20doe',
        {
          'filters': {
            'id': {
              r'$in': [3, 6, 8],
            },
          }
        }: 'filters%5Bid%5D%5B%24in%5D%5B%5D=3&filters%5Bid%5D%5B%24in%5D%5B%5D=6&filters%5Bid%5D%5B%24in%5D%5B%5D=8'
      }.forEach(
        (map, query) {
          test(
            '$map -> $query',
            () => expect(
              mapToQuery(
                map,
                useBrackets: true,
              ),
              query,
              reason: 'legacy brackets',
            ),
          );

          test(
            '$map -> $query',
            () => expect(
              mapToQuery(
                map,
                listFormat: ListFormat.brackets,
              ),
              query,
            ),
          );
        },
      );
    },
  );

  group(
    'mapToQuery maps with comma and nested lists',
    () {
      <Map<String, dynamic>, String>{
        {
          'filters': {
            r'$or': [
              {
                'date': {
                  r'$eq': '2020-01-01',
                }
              },
              {
                'date': {
                  r'$eq': '2020-01-02',
                }
              }
            ],
            'author': {
              'name': {
                r'$eq': 'Kai doe',
              },
            }
          }
        }: 'filters%5B%24or%5D=%7Bdate%3A%20%7B%24eq%3A%202020-01-01%7D%7D%2C%7Bdate%3A%20%7B%24eq%3A%202020-01-02%7D%7D&filters%5Bauthor%5D%5Bname%5D%5B%24eq%5D=Kai%20doe',
        {
          'filters': {
            'id': {
              r'$in': [3, 6, 8],
            },
          }
        }: 'filters%5Bid%5D%5B%24in%5D=3%2C6%2C8'
      }.forEach(
        (map, query) => test(
          '$map -> $query',
          () => expect(
            mapToQuery(
              map,
              listFormat: ListFormat.comma,
              includeNullQueryVars: true,
            ),
            query,
          ),
        ),
      );
    },
  );
```

Reply:
Done.